server/client.go

```diff
@@ -1477,19 +1519,38 @@ func (c *client) flushOutbound() bool {
 	nc.SetWriteDeadline(start.Add(wdl))
 
 	// Actual write to the socket.
-	n, err := nb.WriteTo(nc)
+	n, err := c.out.wnb.WriteTo(nc)
```

Comment:
We want to limit wnb to 8 yes for the actual calls into OS?

Reply:
I think by limiting to 8 buffers before having to call `flushOutbound` again, we're limiting the potential for `writev` to send potentially more than that if it can. The path in `writev` is slightly slower for more than a certain number of buffers (as it requires heap allocations in the kernel) but it still completes in a single context switch regardless, so it's less expensive on syscalls.

Reply:
ok reason I bring it up is seeing us get stuck in certain situations calling in with 1000s of entries and like 100s of MBs of data. I think we want to avoid that maybe if we can.

Reply:
Yep, good point. I'll have a think on how best to handle that.